infer/resource.go

```diff
@@ -933,7 +942,6 @@ func callCustomCheck[T any](
 //
 // It also adds defaults to inputs as necessary, as defined by [Annotator.SetDefault].
 func DefaultCheck[I any](ctx context.Context, inputs resource.PropertyMap) (I, []p.CheckFailure, error) {
-	inputs = applySecrets[I](inputs)
```

Comment:
Thoughts on maybe leaving this as is? Doing it twice is no harm? 

Perhaps as an undocumented behavior of DecodeCheck it's ok to remove it though.

Reply:
The caller of `DefaultCheck` only witnesses the effects of `applySecrets[I]` here when they see values that were previously secret show up as secret again (in the returned value of `Check`). 

This doesn't change observable behavior from the user's perspective (except that `applySecrets[I]` covers values added in `CustomCheck.Check`, the goal of the PR).